circuit_knitting/cutting/cutting_decomposition.py

```diff
@@ -260,3 +267,90 @@ def decompose_observables(
+        optimization: Settings for controlling optimizer behavior. Currently,
+            only a best-first optimizer is supported.
+        constraints: QPU constraints used to generate the cut location search space.
+
+    Returns:
+        A circuit containing :class:`.BaseQPDGate` instances. The subcircuits
+        resulting from cutting these gates will be runnable on the devices
+        specified in ``constraints``.
+
+        A metadata dictionary:
+            cuts: A list of length-2 tuples describing each cut in the output circuit.
+                The tuples are formatted as ``(cut_type: str, cut_id: int)``. The
+                cut ID is the index of the cut gate or wire in the output circuit's
+                ``data`` field.
+            sampling_overhead: The sampling overhead incurred from cutting the specified
+                gates and wires.
+    """
+    circuit_cco = qc_to_cco_circuit(circuit)
+    interface = SimpleGateList(circuit_cco)
+
+    opt_settings = OptimizationSettings.from_dict(optimization)
+
+    # Hard-code the optimization type to best-first
+    opt_settings.set_engine_selection("CutOptimization", "BestFirst")
+
+    constraint_settings = DeviceConstraints.from_dict(constraints)
+
+    # Hard-code the optimizer to an LO-only optimizer
+    optimizer = LOCutsOptimizer(interface, opt_settings, constraint_settings)
+
+    # Find cut locations
+    opt_out = optimizer.optimize()
+
+    wire_cut_actions = []
+    gate_ids = []
+
+    opt_out = cast(DisjointSubcircuitsState, opt_out)
+    opt_out.actions = cast(list, opt_out.actions)
+    for action in opt_out.actions:
+        if action[0].get_name() == "CutTwoQubitGate":
+            gate_ids.append(action[1][0])
+        else:
+            # We only support three types of optimizer cuts
+            assert action[0].get_name() in ("CutLeftWire", "CutRightWire")
+            wire_cut_actions.append(action)
+
+    # First, replace all gates to cut with BaseQPDGate instances.
+    # This assumes each gate to cut is replaced 1-to-1 with a QPD gate.
+    # This may not hold in the future as we stop treating gate cuts individually.
+    circ_out = cut_gates(circuit, gate_ids)[0]
+
+    # Insert all the wire cuts
+    counter = 0
+    for action in sorted(wire_cut_actions, key=lambda a: a[1][0]):
+        inst_id = action[1][0]
+        qubit_id = action[2][0][0] - 1
+        circ_out.data.insert(
+            inst_id + counter,
+            CircuitInstruction(CutWire(), [circuit.data[inst_id].qubits[qubit_id]], []),
+        )
+        counter += 1
+
+    # Return metadata describing the cut scheme
+    sampling_overhead = 1
+    metadata: dict[str, Any] = {"cuts": []}
+    for i, inst in enumerate(circ_out.data):
+        if inst.operation.name == "qpd_2q":
+            sampling_overhead *= inst.operation.basis.overhead
+            metadata["cuts"].append(("Gate Cut", i))
+        if inst.operation.name == "cut_wire":
+            sampling_overhead *= 16
```

Comment:
So it looks like you're recomputing the sampling overhead here. I thought the cut finder already did that and that you could just access that directly. Or am I missing something here?

Reply:
Short answer: The QPD information isn't available in these CutWire objects

More in-depth answer:
[CutWire](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/blob/ef510cae58f38d4c6090c33649796cbefc50c3e8/circuit_knitting/cutting/instructions/cut_wire.py#L19) is really just a dummy instruction indicating we need to allocate an ancilla qubit and do a [Move](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/blob/ef510cae58f38d4c6090c33649796cbefc50c3e8/circuit_knitting/cutting/instructions/move.py#L19) operation at that location. It doesn't contain QPD information. After it is transformed to a ``Move`` operation, we can "cut" the Move operation, which creates two single qubit ``QPDGates`` with labels ["cut_move"](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/blob/ef510cae58f38d4c6090c33649796cbefc50c3e8/circuit_knitting/cutting/qpd/instructions/qpd_gate.py#L144), one on either side of the wire cuts. These contain the QPD information.

I don't want to have to decompose these gates and allocate ancillas and all that, especially since we can hard-code the overhead for wire cuts trivially.

Reply:
I thought you were asking specifically about the hard-coded value here. I see what you mean now. Let me see if we have the sampling overhead available in one of these variables somewhere 🧐

Reply:
I guess what I was getting at was that `opt_out.upper_bound_gamma()` would return to you the overall `gamma` for the cut scheme that is already computed internally. Isn't that what we are looking for in this context?

Reply:
Yes, I realized what you were getting at after I typed all that out lol

Reply:
OK I implemented this as you suggested